src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```diff
@@ -102,26 +94,13 @@ public SimpleElasticsearchPersistentEntity(TypeInformation<T> typeInformation) {
 		}
 	}
 
-	@Override
-	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
-		context.addPropertyAccessor(new BeanFactoryAccessor());
```

Comment:
Removing access to the ApplicationContext removes functionality. What’s the motivation for this change?

Reply:
The only purpose of implementing `ApplicationContextAware` was to set up the EvaluationContext for the SpEL resolving of index names. But with this change, the context from the base class is used, so the setup is not needed anymore. Keeping implementing the interface would lead to a noop implementation of `setApplicationContext`

Reply:
I see. Without access to `ApplicationContext`, SpEL expressions are no longer able to reference beans from the context. We've seen some examples where table names/collection names referenced config classes that contributed parts of the table/collection name.